docs/src/catalyst_functionality/compositional_modeling.md

```diff
@@ -5,16 +5,47 @@ can construct the earlier repressilator model by composing together three
 identically repressed genes, and how to use compositional modeling to create
 compartments.
 
+## A note on *completeness*
+Catalyst `ReactionSystem` can either be *complete* or *incomplete*. *By default they are created as complete*. Here, only complete `ReactionSystem`s can be used to create the various problem types (e.g. `ODEProblem`). However, only incomplete `ReactionSystem`s can be composed using the features described below. Hence, for compositional modeling, `ReactionSystem` must be created as incomplete, and later set to complete before simulation.
+
+To set a `ReactionSystem` created via the DSL as complete, use the `@incomplete` option:
+```@example ex0
+using Catalyst
+degradation_component = @reaction_network begin
+  @incomplete
+  d, X --> 0
+end
+```
+Or when created directly, use the `complete = false` argument:
+```@example ex0
+@parameters d
+@variable t
+@species X(t)
+rx = Reaction(d, [X], nothing)
+@named degradation_component = ReactionSystem([rs], t; complete = false)
+```
+We can test whether a system is complete using the `ModelingToolkit.iscomplete` function:
+```@example ex0
+ModelingToolkit.iscomplete(degradation_component)
+```
+To make a incomplete system complete, we can use the `complete` function:
+```@example ex0
+degradation_component_complete = complete(degradation_component)
+ModelingToolkit.iscomplete(degradation_component_complete)
+```
+
```

Comment:
If we change completeness around, this will have to change. I also have an in-depth introduction in a local branch that goes through stuff like this. Basically, once all of this is over things like this should/will get rewritten anyway.